prow/cmd/image-builder/buildcontroller.go

```diff
@@ -0,0 +1,689 @@
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "watch build pods")
+	}
+
+	// Watch image-builder pod
+	err = c.Watch(
+		&source.Kind{Type: &corev1.Pod{}},
+		&handler.EnqueueRequestForObject{},
+		predicate.NewPredicateFuncs(func(o client.Object) bool {
+			if o.GetName() == r.imageBuilderPod.Name && o.GetNamespace() == r.imageBuilderPod.Namespace {
+				return true
+			}
+			return false
+		}),
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "watch image-builder pod")
+	}
+
+	return r, nil
+
+}
+
+// indexOwnerReferences indexes resources by the UIDs of their owner references.
+func indexOwnerReferences(o client.Object) (refs []string) {
+	for _, ref := range o.GetOwnerReferences() {
+		refs = append(refs, string(ref.UID))
+	}
+	return refs
+}
+
+func (r *buildReconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+
+	// reconcile image-builder pod only
+	if request.NamespacedName != r.imageBuilderPod {
+		r.log.Debugf("Skip reconciliation of pod %v - not the image-builder pod", request.NamespacedName)
+		return reconcile.Result{}, nil
+	}
+
+	// Error handling - try maxErrors times to recover before canceling the build
+	var err error
+	errorHandler := func() {
+		if err == nil {
+			return
+		}
+
+		r.errorCount++
+		if r.errorCount > maxErrors {
+			r.log.Error("Too many errors, stopping build")
+			r.stop(err)
+		}
+
+	}
+	defer errorHandler()
+
+	// Get image-builder pod
+	var ibPod corev1.Pod
+	err = r.client.Get(ctx, request.NamespacedName, &ibPod)
+	if err != nil {
+		r.log.WithError(err).Error("Could not get image-builder pod - this should not happen")
+		return reconcile.Result{}, err
+	}
+
+	err = r.ensureBuildPodDefinition(ctx, &ibPod)
+	if err != nil {
+		r.log.WithError(err).Error("Error ensuring build pod definition")
+		return reconcile.Result{}, err
+	}
+
+	err = r.reconcileBuildPods(ctx, &ibPod)
+	if err != nil {
+		r.log.WithError(err).Error("Error reconcile build pods")
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{RequeueAfter: time.Second * 30}, nil
+}
+
+func (r *buildReconciler) stop(err error) {
+	if !r.canceled {
+		if err != nil {
+			r.err = err
+		}
+		r.cancelFunc()
+		r.canceled = true
+	}
+}
+
+func (r *buildReconciler) ensureBuildPodDefinition(ctx context.Context, ibPod *corev1.Pod) error {
+	if len(r.buildPods) != 0 {
+		r.log.Debug("Build pods defined - skip")
+		return nil
+	}
+	r.log.Info("Start defining build pods")
+
+	pvc := &corev1.PersistentVolumeClaim{}
+	// Use a PVC with the same name and namespace as the image-builder pod
+	err := r.client.Get(ctx, types.NamespacedName{Namespace: ibPod.Namespace, Name: ibPod.Namespace}, pvc)
+	if k8serrors.IsNotFound(err) {
+		r.log.Info("Creating PVC for image build pods")
+		pvc, err = r.createPVC(ctx, ibPod)
+		if err != nil {
+			return errors.Wrap(err, "create PVC")
+
+		}
+	} else if err != nil {
+		return errors.Wrap(err, "get PVC")
+	}
+
+	err = r.defineBuildPods(ibPod, pvc)
```

Comment:
You can use the pod name instead :)
```suggestion
	err = r.defineBuildPods(ibPod)
```